src/commands/query.mjs

```diff
@@ -0,0 +1,121 @@
+//@ts-check
+
+import { container } from "../cli.mjs";
+import {
+  commonConfigurableQueryOptions,
+} from "../lib/command-helpers.mjs";
+import { formatError, formatQueryResponse } from "../lib/fauna-client.mjs";
+
+function validate(argv) {
+  const { existsSync, accessSync, constants } = container.resolve("fs");
+  const dirname = container.resolve("dirname");
+
+  if (argv.input && argv.query) {
+    throw new Error("Cannot specify both --input and --query");
+  }
+
+  if (!argv.input && !argv.query) {
+    throw new Error("No query specified. Pass <query> or --input.");
+  }
+
+  if (argv.input && !existsSync(argv.input)) {
+    throw new Error(`File passed to --input does not exist: ${argv.input}`);
+  }
+
+  if (argv.output) {
+    const outputDir = dirname(argv.output);
+    try {
+      accessSync(outputDir, constants.W_OK);
+    } catch (e) {
+      throw new Error(`Unable to write to output directory: ${outputDir}`);
+    }
+  }
+}
+
+const resolveInput = (argv) => {
+  const { readFileSync } = container.resolve("fs");
+  const logger = container.resolve("logger");
+
+  if (argv.input) {
+    logger.debug(`reading query from ${argv.input}`, "argv");
+    return readFileSync(argv.input, "utf8");
+  }
+
+  if (argv.query === "-") {
+    logger.debug("reading query from stdin", "argv");
+    return readFileSync(process.stdin.fd, "utf8");
+  }
+
+  logger.debug("no --input specified, using <query>", "argv");
+  return argv.query;
+}
+
+async function query(argv) {
+  // validate the arguments and throw if they are invalid
+  validate(argv);
+
+  // resolve the input
+  const expression = resolveInput(argv);
+
+  // get the query handler and run the query
+  try {
+    const results = await container.resolve("runQueryFromString")(expression, argv);
+    const output = formatQueryResponse(results, argv);
+
+    if (argv.output) {
+     container.resolve("fs").writeFileSync(argv.output, output);
+    } else {
+      container.resolve("logger").stdout(output);
+    }
+
+    return results;
+  } catch (err) {
+    err.message = formatError(err, { apiVersion: argv.apiVersion, extra: argv.extra }); 
+    throw err;
+  }
+}
+
+function buildQueryCommand(yargs) {
+  return yargs
+    .positional("query", {
+      type: "string",
+      description: "the query to run; use - to read from stdin",
+    })
+    .nargs('query', 1)
```

Comment:
this'll probably cause issues with queries that have spaces in them. is the behavior better if we don't specify a number and let yargs infer? or do we want to enforce string quoting?

see queries like:
```
fauna query let x = 5 --no-color
```

Reply:
Well, trying to write FQL in zsh without quotes is super dangerous...so I might not do anything about this until we get more feedback from folks.